In [1]:
import pandas as pd 
import numpy as np 
from scipy import stats 
from scipy.stats import norm

In [2]:
#load the dataset 
data = pd.read_csv(r"C:\Users\newpa\Documents\excelr\scaredynewassin\Cutlets.csv")
#display 1st five rows of the datafile 
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

Step 1: Define null hypothesis H0: mean1 = mean2 i.e. There is no difference between the diameters of the two units. 

Step 2: Hence, alternate hypothesis H1: mean1 != mean2 i.e. There is a significant difference between the diameters of the two units.

Step 3: Use the 2 sample t-test for testing out the hypothesis.

In [3]:
data.describe()

Unit A     Unit B
count  36.000000  36.000000
mean    7.019091   6.964297
std     0.284259   0.338459
min     6.437600   6.038000
25%     6.835450   6.756500
50%     6.969500   6.952099
75%     7.279425   7.181900
max     7.516900   7.545900

In [4]:
A = data.iloc[:,0]
A

0     6.809000
1     6.437600
2     6.915700
3     7.301200
4     7.448800
5     7.387100
6     6.875500
7     7.062100
8     6.684000
9     6.823600
10    7.393000
11    7.516900
12    6.924600
13    6.925600
14    6.579700
15    6.839400
16    6.597000
17    7.270500
18    7.282800
19    7.349500
20    6.943800
21    7.156000
22    6.534100
23    7.285400
24    6.995200
25    6.856800
26    7.216300
27    6.680100
28    6.943100
29    7.085200
30    6.779400
31    7.278300
32    7.156100
33    7.394300
34    6.940500
35    7.019091
Name: Unit A, dtype: float64

In [5]:
B = data.iloc[:,1]
B

0     6.770300
1     7.509300
2     6.730000
3     6.787800
4     7.152200
5     6.811000
6     7.221200
7     6.660600
8     7.240200
9     7.050300
10    6.881000
11    7.405900
12    6.765200
13    6.038000
14    7.158100
15    7.024000
16    6.667200
17    7.431400
18    7.307000
19    6.747800
20    6.888900
21    7.422000
22    6.521700
23    7.168800
24    6.759400
25    6.939900
26    7.013300
27    6.918200
28    6.334600
29    7.545900
30    7.099200
31    7.118000
32    6.696500
33    6.578000
34    7.387500
35    6.964297
Name: Unit B, dtype: float64

In [6]:
p_value = stats.ttest_ind(A,B)
p_value

Ttest_indResult(statistic=0.7438259311727656, pvalue=0.4594713250546759)

In [7]:
p_value[1]

0.4594713250546759

In [8]:
print(p_value[1] > 0.05) #compare p_value[1] with α = 0.05 (at 5% significance level)

True


# Conclusion 

As p_value of 0.4722 is greater than α = 0.05 ; we accept null hypothesis. 

Therefore, mean1 = mean2 which means there is no significant difference in diameters of the cutlets between the two units.

# Question 2

In [9]:
data2 = pd.read_csv(r"C:\Users\newpa\Documents\excelr\scaredynewassin\LabTAT.csv")
data2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [10]:
data2.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

ANOVA test statistics should be used here because there are more than 2 samples or columns. 

Null hypothesis H0 : mean1 = mean2 = mean3 = mean4

Alternate hypothesis Ha : If any mean is not equal to the other

In [11]:
# ANOVA ftest : stats.f_oneway(col1,col2,col3,col4)
p_value = stats.f_oneway(data2.iloc[:,0],data2.iloc[:,1],data2.iloc[:,2],data2.iloc[:,3])
print("The p-value is ",p_value[1])

The p-value is  2.1156708949992414e-57


In [12]:
print(p_value[1] > 0.05) # compare it with α = 0.05

False


# Conclusion 

Hence, the p-value for this dataset is less than 0.05 ie the 5% significance level.

Therefore, we can reject the null hypothesis which means at least one sample mean is different from the others. 

Thus, there is difference between the TAT(turnaround time) of the different labs. 

# Question 3

In [13]:
data3 = pd.read_csv(r"C:\Users\newpa\Documents\excelr\scaredynewassin\BuyerRatio.csv")
data3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

 Null hypothesis H0: male-female buyer ratios are same , they do not vary and are not related. 

 Alternate hypothesis Ha: male-female buyer ratios are not same, they vary and are significantly/somewhat related. 

In [14]:
from scipy.stats import chi2_contingency 

In [15]:
obs_values = np.array([[50,142,131,70],[435,1523,1356,750]])
obs_values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [16]:
chi2_contingency(obs_values) #this gives output (chi2 stats value, p-value, df, expected observations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [17]:
chi2_contingency(obs_values)[0]

1.595945538661058

In [18]:
chi2_contingency(obs_values)[1]

0.6603094907091882

In [19]:
print(chi2_contingency(obs_values)[1] > 0.05) #this is the p-value compared to α = 0.05 (5% significance level)

True


# Conclusion

As the p-value (0.6603) is greater than α = 0.05, we accept the Null hypothesis. 

Therefore, the male-female buyer ratios are same across regions and they are not related. 

# Question 4

Null hypothesis H0: Defective % does not vary by center. 

Alternate hypothesis Ha: Defective % varies by center. 

In [21]:
data4 = pd.read_csv(r"C:\Users\newpa\Documents\excelr\scaredynewassin\Costomer+OrderForm.csv")
data4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [23]:
data4.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [24]:
data4.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [25]:
data4.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [26]:
data4.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [27]:
#create contingency table
obs_values = np.array([[271,267,269,280],[29,33,31,20]])
obs_values

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [29]:
chi2_contingency(obs_values) # this gives output: (chi2 stats value, p-value, df, expected observations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [31]:
print("The p_value is: ",chi2_contingency(obs_values)[1])

The p_value is:  0.2771020991233135


In [32]:
print(chi2_contingency(obs_values)[1] > 0.05) #this checks the obtained p-value is greater than 0.05 or not. 

True


# Conclusion

As the p-value (0.2777) is greater than α = 0.05, we accept the Null hypothesis.

Therefore, defective % does not vary across centers. 